src/components/Header/SpacesDropdownMenuItem.tsx

```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import { Space } from '@gitbook/api';
+import { useSelectedLayoutSegment } from 'next/navigation';
+
+import { DropdownMenuItem } from './Dropdown';
+
+function useVariantSpaceHref(variantSpace: Space) {
+    const currentPathname = useSelectedLayoutSegment() ?? '';
+    const targetUrl = new URL(variantSpace.urls.published ?? variantSpace.urls.app);
+    targetUrl.pathname += `/${currentPathname}`;
+    targetUrl.pathname = targetUrl.pathname.replace(/\/{2,}/g, '/').replace(/\/$/, '');
+
+    return targetUrl.toString();
+}
+
+export function SpacesDropdownMenuItem(props: { variantSpace: Space; currentSpace: Space }) {
```

Comment:
@spastorelli keep in mind that when you do a client component like this, all props will have to be serialized in the HTML. So instead of passing the whole object when you use only 1/2 properties, instead pass the few properties as dedicated props.

In that case `active`, `targetUrl` could have been computed in the parent servr component.

Reply:
I can definitely reduce the props to only what's needed from space and variantSpace instead of the full object but not sure about `targetUrl`.

Reply:
actually nevermind sorry I just realized that by targetUrl you meant the value of `variantSpace.urls.published ?? variantSpace.urls.app` 🤦 Sorry about this

Reply:
@SamyPesse https://github.com/GitbookIO/gitbook/pull/2373